src/Document.ts

```diff
@@ -36,25 +36,81 @@ class Document {
 	/** Indicates whether this document is a subdocument of a composing parent */
 	readonly #isSubdocument: boolean;
 
-	public constructor(schema: Schema | null, options: DocumentConstructorOptions) {
+	protected constructor(schema: Schema | null, options: DocumentConstructorOptions) {
 		const { data = {}, record, isSubdocument = false } = options;
 
 		this.#schema = schema;
-		this.#record = [];
+		this.#record = record ?? [];
 		this.#isSubdocument = isSubdocument;
-		this.transformationErrors = [];
+		this._transformationErrors = [];
 
 		Object.defineProperties(this, {
-			transformationErrors: { configurable: false, enumerable: false, writable: false },
+			_transformationErrors: { configurable: false, enumerable: false, writable: false },
 		});
 
-		if (record != null) {
-			this.#transformRecordToDocument(record);
-		}
+		this.#transformRecordToDocument();
+
 		// load the data passed to constructor into document instance
 		assignIn(this, data);
 	}
 
+	/** Create a new Subdocument instance from a record array */
+	public static createSubdocumentFromRecord(schema: Schema, record: MvRecord): Document {
+		return new Document(schema, { record, isSubdocument: true });
+	}
+
+	/** Create a new Subdocument instance from data */
+	public static createSubdocumentFromData(schema: Schema, data: GenericObject): Document {
+		return new Document(schema, { data, isSubdocument: true });
+	}
+
+	/** Create a new Document instance from a record string */
+	public static createDocumentFromRecordString(
+		schema: Schema,
+		recordString: string,
+		dbServerDelimiters: DbServerDelimiters,
+	): Document {
+		const record = Document.convertMvStringToArray(recordString, dbServerDelimiters);
+
+		return new Document(schema, { record });
+	}
+
+	/** Convert a multivalue string to an array */
+	public static convertMvStringToArray(
```

Comment:
Ideally this would have been `protected` as there is no use case for consumers outside this class.  However, there is an open TS issue where `private` and `protected` cannot be used in class constructors that are assigned to variables/returned from functions.  Therefore, this has been made `public` although there is no known external use case.